src/bootstrap/toolstate.rs

```diff
@@ -215,13 +215,18 @@ impl Step for ToolStateCheck {
                             tool, old_state, state
                         );
                     } else {
+                        // This warning only appears in the logs, which most
+                        // people won't read. It's mostly here for testing and
+                        // debugging.
                         eprintln!(
                             "warning: Tool `{}` is not test-pass (is `{}`), \
                             this should be fixed before beta is branched.",
                             tool, state
                         );
                     }
                 }
+                // publish_toolstate.py will be responsible for creating
+                // comments/issues warning people if there is a regression.
```

Comment:
Maybe add that it will also update the "current toolstate" in `latest.json`, and that all of this happens in a separate CI job triggered when bors forwards master to auto because we only want to do that after it became clear that this will land on master.

Reply:
Updated, and added a few other comments.